buildenv/jenkins/docker-slaves/x86/centOS/Dockerfile

```diff
@@ -0,0 +1,219 @@
+# Exception [1] and GNU General Public License, version 2 with the
+# OpenJDK Assembly Exception [2].
+#
+# [1] https://www.gnu.org/software/classpath/license.html
+# [2] http://openjdk.java.net/legal/assembly-exception.html
+#
+# SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+
+# To use this docker file:
+#   docker build -t=openj9 .
+#   docker run -it openj9
+
+FROM centos:centos6.9
+
+# Install required OS tools
+
+ENV USER="jenkins"
+
+RUN yum -y install \
+    alsa-lib-devel \
+    autoconf \
+    bind-utils \
+    bison \
+    bzip2 \
+    ca-certificates \
+    cups-devel \
+    cmake \
+    cpio \
+    elfutils-libelf-devel \
+    file-devel \
+    file-libs \
+    flex \
+    fontconfig \
+    fontconfig-devel \
+    freetype-devel \
+    gettext \
+    glibc \
+    glibc-common \
+    glibc-devel \
+    gmp-devel \
+    java-1.7.0-openjdk\
+    java-1.8.0-openjdk\
+    java-1.7.0-openjdk-devel \
+    java-1.8.0-openjdk-devel \
+    lbzip2 \
+    libdwarf \
+    libX11-devel \
+    libXext-devel \
+    libXrender-devel \
+    libXt-devel \
+    libXtst-devel \
+    make \
+    mesa-libGL-devel \
+    mpfr-devel \
+    numactl-devel \
+    ntp \
+    openssl-devel \
+    openssh-server \
+    openssh-clients \
+    perl-CPAN \
+    perl-DBI \
+    perl-devel \
+    perl-GD \
+    perl-libwww-perl \
+    perl-Time-HiRes \
+    unzip \
+    vim \
+    wget\
+    xz \
+    zip \
+  && rm -rf /var/lib/apt/lists/*
+
+RUN yum -y install https://$(rpm -E '%{?centos:centos}%{!?centos:rhel}%{rhel}').iuscommunity.org/ius-release.rpm \
+  && yum -y install libdwarf-devel
+
+RUN /usr/bin/perl -MCPAN -e'install Text::CSV'
+
+RUN cd /etc/yum.repos.d \
+  && wget http://people.centos.org/tru/devtools-2/devtools-2.repo -O /etc/yum.repos.d/devtools-2.repo \
+  && yum -y install devtoolset-2-gcc devtoolset-2-binutils \
+  && yum -y install devtoolset-2-gcc-c++ devtoolset-2-gcc-gfortran \
+  && scl enable devtoolset-2 bash
+
+RUN mv /opt/rh/devtoolset-2/root/usr/bin/gcc /opt/rh/devtoolset-2/root/usr/bin/gcc-4.8 \
+  && mv /opt/rh/devtoolset-2/root/usr/bin/g++ /opt/rh/devtoolset-2/root/usr/bin/g++-4.8 \
+  && ln -s /opt/rh/devtoolset-2/root/usr/bin/gcc-4.8 /usr/bin/gcc \
+  && ln -s /opt/rh/devtoolset-2/root/usr/bin/g++-4.8 /usr/bin/g++ \
+  && ln -s /opt/rh/devtoolset-2/root/usr/bin/gcc-4.8 /usr/bin/cc
+
+RUN yum -y install curl-devel expat-devel gettext-devel openssl-devel zlib-devel perl-ExtUtils-MakeMaker\
+  && cd /usr/src \
+  && wget https://www.kernel.org/pub/software/scm/git/git-2.5.3.tar.gz \
+  && tar xzf git-2.5.3.tar.gz \
+  && rm git-2.5.3.tar.gz \
+  && cd git-2.5.3 \
+  && make prefix=/usr/local/git all \
+  && make prefix=/usr/local/git install \
+  && echo 'pathmunge /usr/local/git/bin/' > /etc/profile.d/git.sh \
+  && chmod +x /etc/profile.d/git.sh \
+  && ln -s /usr/local/git/bin/git /usr/bin/git
```

Comment:
why is this symlink needed? once `git` is installed, a user should directly be able to call `git` from cmdline.

Reply:
I don't think the `pathmunge` is needed and its specific to RPM based OSs so we can remove it and the `chmod` but we still need the symlink.